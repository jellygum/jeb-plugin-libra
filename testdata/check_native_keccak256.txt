CompiledProgram: {
Modules: [
],
Script: CompiledScript: {
Main:
	public 0x0.<SELF>.main(): ()
		locals(0): ByteArray, ByteArray, ByteArray,
		LdByteArray(0)
		StLoc(0)
		CopyLoc(0)
		Call(0x0.Hash.keccak256(ByteArray): (ByteArray))
		StLoc(1)
		LdByteArray(1)
		StLoc(2)
		LdConst(42)
		MoveLoc(1)
		MoveLoc(2)
		Eq
		Assert
		Ret
Struct Handles: []
Module Handles: [
	0x0.<SELF>,
	0x0.Hash,]
Function Handles: [
	0x0.<SELF>.main(): (),
	0x0.Hash.keccak256(ByteArray): (ByteArray),]
Type Signatures: []
Function Signatures: [
	(): (),
	(ByteArray): (ByteArray),]
Locals Signatures: [
	ByteArray, ByteArray, ByteArray,]
Strings: [
	<SELF>,
	Hash,
	main,
	keccak256,]
ByteArrays: [
	0x616263,
	0x4e03657aea45a94fc7d47ba826c8d667c0d1e6e33a64a036ec44f58fa12d6c45,]
Addresses: [
	0x0,]
}

}
