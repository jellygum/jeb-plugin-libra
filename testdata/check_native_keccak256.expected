Module Handles: [
    0x0.<SELF>,
    0x0.Hash,
]
Struct Handles: [
]
Function Handles: [
    0x0.<SELF>.main(): (),
    0x0.Hash.keccak256(bytearray): (bytearray),
]
Struct Definitions: [
]
Field Definitions: [
]
Type Signatures: [
]
Function Signatures: [
    (): (),
    (bytearray): (bytearray),
]
Local Signatures: [
    bytearray, bytearray, bytearray,
]
Strings: [
    <SELF>,
    Hash,
    main,
    keccak256,
]
ByteArrays: [
    616263,
    4E03657AEA45A94FC7D47BA826C8D667C0D1E6E33A64A036EC44F58FA12D6C45,
]
Addresses: [
    0x0,
]
Function Definitions: [
]
public 0x0.<SELF>.main(): ()
    max_stack_size: 4
    locals: bytearray, bytearray, bytearray
    LD_BYTEARRAY IdxByteArray(0)       [0,1]
    ST_LOC IdxLocal(0)                 [1,0]
    COPY_LOC IdxLocal(0)               [0,1]
    CALL IdxFuncHandle(1)              [1,1]
    ST_LOC IdxLocal(1)                 [1,0]
    LD_BYTEARRAY IdxByteArray(1)       [0,1]
    ST_LOC IdxLocal(2)                 [1,0]
    LD_CONST ImmUint64(42)             [0,1]
    MOVE_LOC IdxLocal(1)               [1,2]
    MOVE_LOC IdxLocal(2)               [2,3]
    EQ                                 [3,2]
    ASSERT                             [2,0]
    RET                                [0,0]

